### ISSUE TITLE
Some dynamic patterns don't work together

### ISSUE THREAD

**Issue by vladdu** [mrmlnc/fast-glob#316]
### Environment

  * OS Version: Ubuntu 20.04	
  * Node.js Version: 14.17

### Actual behavior

Using dynamic patterns that start with ../ together with other dynamic patterns that will be put into the '.' group, doesn't work. All patterns are put inside the '.' group and my guess is that the file system walker won't go outside of the current directory and the ../ patterns will get ignored.

For example, 
```js
[
  "{a,b}/**/*.md",
  "../**/*.html",    // these get ignored
]
```
```js
[
  "a/**/*.md",
  "b/**/*.md",
  "../**/*.html",    // these get ignored
]
```

### Expected behavior

All patterns are used.

### Steps to reproduce

I am using this from a third party application, so I'm not sure how to reproduce other than creating patterns similar to the above that match something on your disk.

**Comment by vladdu** [comment]
I created a simple PR #317 that fixes this issue, but I'm not sure if it affects other things.

**Comment by mrmlnc** [comment]
At first glance, we must pick the patterns with the `..` base directory to a separate group.

**Comment by vladdu** [comment]
The solution that minimizes the amount of file walking is to compare all patterns and see which ones include others and pick the topmost one as base for the group. So in our example, the base dir would be ../

**Comment by vladdu** [comment]
Are absolute paths patterns allowed? In that case, just checking for ./ and ../ and ../ won't be enough, there can be collisions anyway, and let's not even mention symlinks. I suppose that if absolute paths are used, the user can accept the responsibility that there might be not optimally efficient (if it's documented) and it might be enough to consider only the relative patterns.

**Comment by mrmlnc** [comment]
Fixed by #320. Will be shipped on this week.
